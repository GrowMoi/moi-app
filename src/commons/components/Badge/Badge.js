import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const Circle = styled(View)`
  border-radius: ${props => props.size / 2};
  background-color: ${props => props.badgeColor && props.badgeColor};
  padding-horizontal: 2;
  padding-vertical: 2;
  min-width: ${props => props.size};
  min-height: ${props => props.size};
  justify-content: center;
  align-items: center;
`;

const Counter = styled(Text)`
  background-color: transparent;
  color: ${props => props.numberColor && props.numberColor};
  font-weight: bold;
  font-size: ${props => (props.size * 0.55)};
`;

const Badge = ({ size, value, badgeColor, numberColor }) => (
  <Circle size={size} badgeColor={badgeColor}>
    <Counter size={size} numberColor={numberColor}>{value}</Counter>
  </Circle>
);

Badge.defaultProps = {
  size: 25,
  value: 0,
  badgeColor: 'red',
  numberColor: 'white',
};

Badge.propTypes = {
  size: PropTypes.number,
  value: PropTypes.number,
  number: PropTypes.number,
  badgeColor: PropTypes.string,
  numberColor: PropTypes.string,
};

export default Badge;
