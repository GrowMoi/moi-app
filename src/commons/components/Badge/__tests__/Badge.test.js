import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import Badge from '../Badge';

describe('Badge', () => {
  it('should render expected', () => {
    const wrapper = renderer.create(<Badge />);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
