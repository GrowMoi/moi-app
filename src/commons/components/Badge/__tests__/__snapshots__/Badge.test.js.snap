// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render expected 1`] = `
<View
  badgeColor="red"
  size={25}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 12.5,
        "justifyContent": "center",
        "minHeight": 25,
        "minWidth": 25,
        "paddingHorizontal": 2,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberColor="white"
    size={25}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 13.750000000000002,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    0
  </Text>
</View>
`;
